version: "3.7"

services:
  app_proxy:
    environment:
      APP_HOST: $APP_THUNDERHUB_IP
      APP_PORT: $APP_THUNDERHUB_PORT

  web:
    image: apotdevin/thunderhub:v0.13.14@sha256:6ee592a8335ec203f9c7056ddd22f286f8135b5888005d84db70787a91e3e72f
    # We now have to run as root to avoid schema.gql permission errors
    # user: "1000:1000"
    restart: on-failure
    stop_grace_period: 1m
    volumes:
      - ${APP_LIGHTNING_NODE_DATA_DIR}:/lnd:ro
      - ${APP_DATA_DIR}/data:/data
    environment:
      MASTER_PASSWORD_OVERRIDE: "$APP_PASSWORD"
      NO_VERSION_CHECK: "true"
      LOG_LEVEL: "debug"
      ACCOUNT_CONFIG_PATH: "/data/thubConfig.yaml"
      YML_ENV_1: "$APP_LIGHTNING_NODE_IP:$APP_LIGHTNING_NODE_GRPC_PORT"
      TOR_PROXY_SERVER: "socks://${APP_ID}.tor:9050"
    networks:
      default:
        ipv4_address: $APP_THUNDERHUB_IP
  
  tor:
    image: getumbrel/tor:0.4.7.8@sha256:2ace83f22501f58857fa9b403009f595137fa2e7986c4fda79d82a8119072b6a
    user: "1000:1000"
    restart: on-failure
    volumes:
      - ${APP_DATA_DIR}/torrc:/etc/tor/torrc:ro
      - ${TOR_DATA_DIR}:/data
    environment:
      HOME: "/tmp"
    hostname: "${APP_ID}.tor"